src/dialogs/more-info/ha-more-info-history.ts

```diff
@@ -42,19 +45,26 @@ export class MoreInfoHistory extends LitElement {
 
   protected updated(changedProps: PropertyValues): void {
     super.updated(changedProps);
-    if (!this.entityId) {
-      clearInterval(this._historyRefreshInterval);
-    }
 
     if (changedProps.has("entityId")) {
       this._stateHistory = undefined;
 
-      this._getStateHistory();
+      this._throttleHistoryFunction();
+      return;
+    }
+
+    if (!this.entityId || !changedProps.has("hass")) {
```

Comment:
What if the entity changes to undefined, will it try to get the state history since we don't return until after the throttle is called